behave/features/steps/foxadmin.py

```diff
@@ -0,0 +1,80 @@
+from behave import *
+from selenium.webdriver.common.by import By
+import datetime
+from selenium.webdriver.support.ui import WebDriverWait
+import os
+from features.constants import MINIMUM_WAIT_UNTIL_TIME
+
+
+@step(u'I enter a date range')
+def step_impl(context):
+    # report can only span 8 days
+    # this can use the callbacks created for the other tests as all it does is check they are there
+    # no concerns about order of tests running as it uses background task to create them if they are not there
+    date_from = (datetime.datetime.now() + datetime.timedelta(days=-1)).strftime("%d/%m/%Y")
+    date_to = (datetime.datetime.now() + datetime.timedelta(days=1)).strftime("%d/%m/%Y")
+    context.helperfunc.find_by_name("date_from").send_keys(date_from)
+    context.helperfunc.find_by_name("date_to").send_keys(date_to)
+
+
+@step(u'I select \'Export\'')
+def step_impl(context):
+    # no point checking for green message as it appears and stays even if this button clicked many times
+    # but worth remembering how many rows in the table before we start
+    xpath = "//div[@class='report-exports']/table/tbody/tr"
+    # this will return an empty list if there is no table to be found
+    context.how_many_reports_exist = len(context.helperfunc.driver().find_elements_by_xpath(xpath))
+    context.helperfunc.click_button(By.NAME, "action")
+
+
+@step(u'the report is processed and available to download as a .csv')
+def step_impl(context):
+    # check that there is a new line in the table for this export
+    # wait until the row created
+    xpath = "//div[@class='report-exports']/table/tbody/tr"
+
+    class WaitForReportToBeDownloaded(object):
```

Comment:
I think these `classes` should be in its own file together and imported in. Would allow them to be reused somewhere else if required. Would also help reduce the size of this method.

Reply:
Maybe, but then I don't know if they are useable elsewhere. I can take them out and put them somewhere else other than the method.

Reply:
Okay that makes sense. Agreed, maybe create separate steps to reduce the size then.